src/Features/VisualBasic/Portable/Completion/CompletionProviders/SymbolCompletionProvider.ItemRules.vb

```diff
@@ -23,6 +23,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
                 Return MyBase.IsCommitCharacter(completionItem, ch, textTypedSoFar)
             End Function
 
+            Public Overrides Function IsBetterPreselectedMatch(item As CompletionItem, other As CompletionItem, textTypedSoFar As String) As Boolean?
+                Dim symbolItem = DirectCast(item, SymbolCompletionItem)
+                Dim otherSymbolItem = DirectCast(other, SymbolCompletionItem)
+                If TypeOf otherSymbolItem.Symbols.First() Is ITypeSymbol Then
+                    Return symbolItem.Symbols.First().MatchesKind(SymbolKind.Event, SymbolKind.Field, SymbolKind.Local, SymbolKind.Method,
```

Comment:
I view the enum (/integer/byte/etc) as the mechanism for coordination between providers with IsBetterPreselectedMatch letting a provider determine which of its items is better. It's possible that we should start out with more priority values, but I wouldn't want extenders (or ourselves) to struggle with unexpected behavior if we conflate the inter-provider coordination with the intra-provider coordination.